[mlir][EmitC] Add member access ops

marbre opened the pull request:
This adds an `emitc.member` and `emitc.member_of_ptr` operation for the corresponding member access operators. Furthermore, `emitc.assign` is adjusted to be used with the member acces operators.

aniragil commented on mlir/lib/Target/Cpp/TranslateToCpp.cpp:
```diff
@@ -183,6 +183,12 @@ struct CppEmitter {
   // Returns the textual representation of a subscript operation.
   std::string getSubscriptName(emitc::SubscriptOp op);
 
+  // Returns the textual representation of a member (of object) operation.
+  std::string getOperandAndMemberName(emitc::MemberOp op);
+
+  // Returns the textual representation of a member of pointer operation.
+  std::string getOperandAndMemberName(emitc::MemberOfPtrOp op);
+
```
Name is a bit opaque, I'd rename to `createMemberAccess`

marbre replied:
The name is inspired by `getSubscriptName`, but I wasn't really satisfied either with `getOperandAndMemberName`. I renamed to `createMemberAccess` as suggested.

aniragil reviewed (changes requested):
LGTM (with a nit).
This will address [this issue](https://discourse.llvm.org/t/does-emitc-dialect-support-member-access-opeartor/79600), right?

marbre committed:
[mlir][EmitC] Add member access ops

This adds an `emitc.member` and `emitc.member_of_ptr` operation for the
corresponding member access operators. Furthermore, `emitc.assign` is
adjusted to be used with the member acces operators.

marbre commented:
> LGTM (with a nit). This will address [this issue](https://discourse.llvm.org/t/does-emitc-dialect-support-member-access-opeartor/79600), right?

Thanks for the review. This actually solves the problem you are pointing to, although unfortunately I have missed this thread so far. The motivation rather is to get a few more things right before landing @simon-camp's lvalue patch.
